app/services/CommonTaskListService.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import models.get.{AllCISDeductions, CISSource}
+import models.tasklist._
+import uk.gov.hmrc.http.HeaderCarrier
+
+import java.time.Instant
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CommonTaskListService @Inject()(appConfig: AppConfig,
+                                      cisDeductionsService: CISDeductionsService
+                                     ) {
+
+  def get(taxYear: Int, nino: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[TaskListSection] = {
+    cisDeductionsService.getCISDeductions(nino, taxYear).map {
+      case Left(_) => AllCISDeductions(None, None)
+      case Right(cis) => cis
+    }.map { cis =>
+      val tasks: Option[TaskListSectionItem] = getTasks(cis, taxYear)
+      TaskListSection(SectionTitle.CISTitle, tasks)
+    }
+  }
+
+  private def getTasks(cisDeductions: AllCISDeductions, taxYear: Int): Option[TaskListSectionItem] = {
+
+    // TODO: these will be links to the new individual CYA pages when they are made
```

Comment:
You can remove this comment, CIS will not need changes and should load the current CYA page with no problems.